src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/SaveCoreCaseDataService.java

```diff
@@ -55,7 +69,20 @@ public CaseDetails save(
             .data(toJson(ccdCase))
             .build();
 
-        return submit(authorisation, eventRequestData, caseDataContent, represented);
+        CaseDetails caseDetails = submit(authorisation, eventRequestData, caseDataContent, represented);
+
+        if (letterHolderId != null) {
```

Comment:
Will there be cases where null `letterHolderId` will be passed to the `save` method? If yes then I think we should have 2 methods -- e.g. `save` as it is now and `saveWithLetterHolder`, which calls the first one and then performs the letter holder linking logic. I think it's best to avoid methods which have "optional" arguments and require nulls to be passed in. What do you think?

Reply:
its not that, its represented or not currently, I didn't want to check the represented flag as they are building a defendant journey, but if I checked the represented flag would you be happier with that?

Reply:
I'd prefer 2 methods but a boolean will make me slightly happier than with as it is now 🙂